hack/debug-network.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+# echoes args to stderr and exits
+die () {
+    echo "$*" 1>&2
+    exit 1
+}
+
+# echoes the command provided as $@ and then runs it
+echo_and_eval () {
```

Comment:
I'm guessing here, but since you're piping everything coming out of these commands into files, you don't really get to see what's going on, so I can see why you might use this type of construct -- instead, you can keep `std{err,out}` going to the console where someone can watch it and `tee` to a file to collect it at the same time, getting rid of this overhead.